packages/nimble-components/src/wafer-map/modules/zoom-handler.ts

```diff
@@ -6,118 +6,68 @@ import {
+                        * this.wafermap.canvasSideLength,
+                    this.wafermap.dataManager!.containerDimensions.width
+                        * this.wafermap.dataManager!.containerDimensions.height
                 )
             ])
             .translateExtent([
                 this.minExtentPoint,
                 [
-                    this.canvasLength + this.extentPadding,
-                    this.canvasLength + this.extentPadding
+                    this.wafermap.canvasSideLength + this.extentPadding,
+                    this.wafermap.canvasSideLength + this.extentPadding
                 ]
             ])
             .filter((event: Event) => {
-                const transform = zoomTransform(this.canvas);
-                return transform.k >= this.minScale || event.type === 'wheel';
+                const transform = zoomTransform(this.wafermap.canvas);
+                const filterEval = transform.k >= this.minScale || event.type === 'wheel';
+                return filterEval;
             })
-            .on('zoom', (event: { transform: ZoomTransform }) => {
-                const transform = event.transform;
-                const canvasContext = this.canvas.getContext('2d');
-                if (canvasContext === null) {
-                    return;
-                }
-                canvasContext.save();
-                if (transform.k === this.minScale) {
-                    this.zoomTransform = zoomIdentity;
-                    this.clearCanvas(
-                        canvasContext,
-                        this.canvasLength,
-                        this.canvasLength
-                    );
-                    this.scaleCanvas(
-                        canvasContext,
-                        zoomIdentity.x,
-                        zoomIdentity.y,
-                        zoomIdentity.k
-                    );
-                    this.renderingModule.drawWafer(this.zoomTransform.k);
-                    zoomBehavior.transform(
-                        select(this.canvas as Element),
-                        zoomIdentity
-                    );
-                } else {
-                    this.zoomTransform = transform;
-                    this.clearCanvas(
-                        canvasContext,
-                        this.canvasLength * this.zoomTransform.k,
-                        this.canvasLength * this.zoomTransform.k
-                    );
-                    this.scaleCanvas(
-                        canvasContext,
-                        transform.x,
-                        transform.y,
-                        transform.k
-                    );
-                    this.renderingModule.drawWafer(this.zoomTransform.k);
-                }
-                canvasContext.restore();
-                this.zoomContainer.setAttribute(
-                    'transform',
-                    this.zoomTransform.toString()
-                );
+            .on('zoom', (event: ZoomEvent) => {
```

Comment:
I did not understand how you can add the zoom event listeners multiple times without issue but the following docs describe why: https://github.com/d3/d3-zoom/blob/v3.0.0/README.md#zoom_on
Can you please include a comment that describes it?

```suggestion
            // D3 will automatically remove existing handlers when adding new ones
            // See: https://github.com/d3/d3-zoom/blob/v3.0.0/README.md#zoom_on
            .on('zoom', (event: ZoomEvent) => {
```

Reply:
applied